api/src/routes/flashcard.rs

```diff
@@ -88,3 +88,47 @@ pub async fn get_flashcard(
 
     HttpResponse::Ok().json(flashcards)
 }
+
+exportable! {
+    pub struct FlashcardFromIDGet {
+        flashcard_id: i32
+    }
+}
+
+exportable! {
+    pub struct FlashcardFromIDResp {
+        id: i32,
+        folder_id: i32,
+        term: String,
+        definition: String
+    }
+}
+
+#[get("/flashcard/fromid")]
+pub async fn get_flashcard_from_id(
+    data: Data<AppState>,
+    info: web::Query<FlashcardFromIDGet>,
+    req: HttpRequest,
+) -> impl Responder {
+    let user_id: i32 = utils::get_user_id(&req).unwrap();
+    let flashcards = sqlx::query_as!(
+        // TODO: replace "select id where id"
+        FlashcardFromIDResp,
+        "SELECT id, folder_id, term, definition FROM flashcard WHERE id = $1",
+        info.flashcard_id
+    )
+    .fetch_all(data.db_pool.as_ref())
+    .await
+    .expect("Could not find flashcard");
+    let flashcard = flashcards.first().expect("Could not find flashcard");
+
+    if !get_user_permissions(flashcard.folder_id, user_id, &data.db_pool)
```

Comment:
Probably best to do this before the lookup - avoids extra computation for nothing, and currently an "attacker" could find out if a flashcard ID exists or not as it will tell them if it couldn't be found before it checks their permission

Reply:
The folder id from the lookup is required to check the permissions. We could send the same response if the flashcard is invalid or if the user is unauthorised - currently the server just errors. Actually, this endpoint would end up being unnecessary if we somehow cache the flashcards when they're selected.